plugins/by-name/clangd-extensions/default.nix

```diff
@@ -76,18 +76,34 @@ lib.nixvim.plugins.mkNeovimPlugin {
 
   extraConfig = cfg: {
     warnings = lib.nixvim.mkWarnings "plugins.clangd-extensions" {
-      when = !config.plugins.lsp.enable;
+      when =
+        !(
+          (config.plugins.lsp.servers.clangd.enable or false) || (config.lsp.servers.clangd.enable or false)
+        );
 
       message = ''
-        You have enabled `clangd-extensions` but not the lsp (`plugins.lsp`).
-        You should set `plugins.lsp.enable = true` to make use of the clangd-extensions' features.
+        You have enabled `clangd-extensions` but not the clangd lsp
+        (`plugins.lsp.servers.clangd` or `lsp.servers.clangd`). You should set
+        `plugins.lsp.servers.clangd.enable = true` or
+        `lsp.servers.clangd.enable = true` to make use of the
+        clangd-extensions' features.
       '';
     };
 
-    plugins.lsp = {
+    lsp = {
       servers.clangd = {
         enable = lib.mkDefault true;
```

Comment:
I would drop this, as outlined in my other comments since there isn't a `lsp.enable` option (like `plugins.lsp.enable`) this has more impact on the user's config than a plugin typically should.